packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -303,15 +314,15 @@ export class EthModule {
 
   // eth_call
 
-  private _callParams(params: any[]): [RpcCallRequest, OptionalBlockTag] {
-    return validateParams(params, rpcCallRequest, optionalBlockTag);
+  private _callParams(params: any[]): [RpcCallRequest, OptionalRpcNewBlockTag] {
+    return validateParams(params, rpcCallRequest, optionalRpcNewBlockTag);
   }
 
   private async _callAction(
     rpcCall: RpcCallRequest,
-    blockTag: OptionalBlockTag
+    blockTag: OptionalRpcNewBlockTag
   ): Promise<string> {
-    const blockNumberOrPending = await this._resolveBlockTag(blockTag);
+    const blockNumberOrPending = await this._resolveNewBlockTag(blockTag);
```

Comment:
There's now different methods to resolve the block tags for the new and old kind of blocktags. The main difference is that the new one throws if the block is not found. 

The old one doesn't, but return `undefined`. This is like that because the old one is used in places where you explicitly ask for a value related to that tag, and `null` is returned if it's not found.

The new kind is used just as a context of another operation, like `eth_call`.